core/src/forward_packet_batches_by_accounts.rs

```diff
@@ -0,0 +1,343 @@
+use {
+    crate::unprocessed_packet_batches::{self, ImmutableDeserializedPacket},
+    solana_perf::packet::Packet,
+    solana_runtime::{
+        bank::Bank,
+        block_cost_limits,
+        cost_tracker::{CostTracker, CostTrackerError},
+    },
+    solana_sdk::pubkey::Pubkey,
+    std::{rc::Rc, sync::Arc},
+};
+
+/// `ForwardBatch` to have half of default cost_tracker limits, as smaller batch
+/// allows better granularity in composing forwarding transactions; e.g.,
+/// transactions in each batch are potentially more evenly distributed across accounts.
+const FORWARDED_BLOCK_COMPUTE_RATIO: u32 = 2;
+/// this number divided by`FORWARDED_BLOCK_COMPUTE_RATIO` is the total blocks to forward.
+/// To accommodate transactions without `compute_budget` instruction, which will
+/// have default 200_000 compute units, it has 100 batches as default to forward
```

Comment:
this 200k default assumption isn't essential, right? (Doesn't seem to be from these changes).  we're about to activate the feature that makes the default units `min(200k * ix_count, 1.4M)`